src/openrct2/drawing/Sprite.cpp

```diff
@@ -746,39 +756,71 @@ extern "C"
 
         if (image_id < SPR_G2_BEGIN)
         {
-            return &g1Elements[image_id];
+            if (image_id >= (sint32)_g1ElementsCount)
+            {
+                return nullptr;
+            }
+            return &_g1Elements[image_id];
         }
         if (image_id < SPR_CSG_BEGIN)
         {
             const uint32 idx = image_id - SPR_G2_BEGIN;
-            openrct2_assert(idx < _g2.header.num_entries,
-                            "Invalid entry in g2.dat requested, idx = %u. You may have to update your g2.dat.", idx);
+            if (idx >= _g2.header.num_entries)
+            {
+                // TODO I guess we still want to warn the user, but an assert is not very good.
+                //      Logging to stdout isn't ideal either as it would flood the console with
+                //      the same message.
+                // openrct2_assert(idx < _g2.header.num_entries,
+                //     "Invalid entry in g2.dat requested, idx = %u. You may have to update your g2.dat.", idx);
```

Comment:
While I agree with your comment, this assert also helped me occasionally to know what the issue was.

I'd rather you left in the assert or replaced it with log. Yes, the latter might flood, but it is not in a valid state anyway and the only way to fix now-invisible parts is to fix the g2.